be/src/storage/compaction_manager.h

```diff
@@ -174,6 +178,9 @@ class CompactionManager {
 
     std::unique_ptr<ThreadPool> _compaction_pool = nullptr;
     std::thread _scheduler_thread;
+
+    std::mutex _compact_threads_mutex;
```

Comment:
```suggestion
    mutable std::mutex _compact_threads_mutex;
```

Add `mutable` to the mutex so it can be used in a `const` member function.